xlators/features/bit-rot/src/bitd/bit-rot-scrub.c

```diff
@@ -1324,6 +1324,10 @@ br_scrubber_scale_up(xlator_t *this, struct br_scrubber *fsscrub,
                "%d/%d [total scrubber(s): %d]",
                v2, i, diff, (v1 + i));
 
+    if (scrub != NULL) {
+        GF_FREE(scrub);
+    }
```

Comment:
I doubt this is a false positive, by looking at the working of for loop from the line: 1303-1316.

Reply:
False positive ? if the loop exits at line 1312, `scrub` needs to be destroyed.

However, if the loop completes successfully, `scrub` won't be NULL, but it has been added to `fsscrub->scrubbers`, so it must not be destroyed in this case.

Reply:
Maybe what's wrong is the condition at line 1318... and `scrub` needs to be destroyed after `error_return` label.

Reply:
Or, changing the condition at line: 1327 to `if (ret && scrub)` should work as well?

Reply:
ret is also added in the condition check in the latest commit.